website/source/guides/configuration/authentication.html.md

```diff
@@ -0,0 +1,671 @@
+---
+layout: "guides"
+page_title: "AppRole Pull Authentication - Guides"
+sidebar_current: "guides-configuration-authentication"
+description: |-
+  Authentication is a process in Vault by which user or machine-supplied
+  information is verified to create a token with pre-configured policy.
+---
+
+# Authentication
+
+Before a client can interact with Vault, it must authenticate against an [**auth
+backend**](/docs/auth/index.html) to acquire a token. This token has policies attached so
+that the behavior of the client can be governed.
+
+Since tokens are the core method for authentication within Vault, there is a
+**token** auth backend (often refer as **_token store_**). This is a special
+auth backend responsible for creating and storing tokens.
+
+### Auth Backends
+
+Auth backends performs authentication to verify the user or machine-supplied
+information. Some of the supported auth backends are targeted towards users
+while others are targeted toward machines or apps. For example,
+[**LDAP**](/docs/auth/ldap.html) auth backend enables user authentication using
+an existing LDAP server while [**AppRole**](/docs/auth/approle.html) auth
+backend is recommended for machines or apps.
+
+[Getting Started](/intro/getting-started/authentication.html) guide walks you
+through how to enable GitHub auth backend for user authentication.
+
+This introductory guide focuses on generating tokens for machines or apps by
+enabling [**AppRole**](/docs/auth/approle.html) auth backend.
+
+
+## Reference Material
+
+- [Getting Started](/intro/getting-started/authentication.html)
+- [Auth Backends](docs/auth/index.html)
+- [GitHub Auth APIs](/api/auth/github/index.html)
+
+
+## Estimated Time to Complete
+
+10 minutes
+
+## Personas
+
+The end-to-end scenario described in this guide involves two personas:
+
+- **`admin`** with privileged permissions to configure an auth backend
+- **`app`** is the consumer of secrets stored in Vault
+
+
+## Challenge
+
+Think of a scenario where a DevOps team wants to configure Jenkins to read
+secrets from Vault so that it can inject the secrets to app's environment
+variables (e.g. `MYSQL_DB_HOST`) at deployment time.
+
+Instead of hardcoding secrets in each build script as a plaintext, Jenkins
+retrieves secrets from Vault.
+
+As a user, you can authenticate with Vault using your LDAP credentials, and
+Vault generates a token. This token has policies granting you to perform
+appropriate operations.
+
+How can a Jenkins server programmatically request a token so that it can read
+secrets from Vault?  
+
+
+## Solution
+
+Enable **AppRole** auth backend so that the Jenkins server can obtain a Vault
+token with appropriate policies attached. Since each AppRole has attached
+policies, you can write fine-grained policies limiting which app can access
+which path.  
+
+
+## Prerequisites
```

Comment:
should we reference the `policies` guide here too? it gets referenced later but it might as well be stated up front

Reply:
**Policy requirements** section immediately follows because of that.  The Getting Started guide starts the server in "dev" mode and use the root token until mid-way through.  For a beginner, wouldn't it be confusing to mention the policy as a prerequisite (since it's not for root)?

Reply:
makes sense! you're right